### PR TITLE
refactor Sentinel to use `karaf-maven-plugin` assembly only

### PR DESCRIPTION
**Pull request by RangerRick**
This PR is step 1 in the process of cleaning up how our Karaf assemblies for OpenNMS core, minion, and sentinel are built.

A large majority of the work that went into this will apply to (and make easier) the core and minion refactor that still needs doing.

It does a number of things, including cleaning up a number of dependencies, fixing implied dependencies in features files by creating explicit dependencies on other features and bundles, and then completely reworking the Sentinel assembly to do _all_ of the work that was previously done in the assembly and "repository" projects.

For details on the general idea of what this PR is doing, see the "Design (Future)" portion of [the Development Tips](https://github.com/OpenNMS/opennms/blob/foundation-2023/DEVELOPMENT-TIPS.md) document.

### Next Steps

1. Once this is merged, let it get into at least one Horizon 33.x release, so we can make sure there aren't any gotchas.
2. Assuming everything is OK, I'm leaving the [foundation-2024 version of this branch](https://github.com/OpenNMS/opennms/tree/features/karaf-assembly-refactor-NMS-16294) around so you can merge it once you're ready.
3. Eventually, this work is going to have to be "ported" to the Horizon core and minion assemblies as well. I've created [NMS-16413](https://opennms.atlassian.net/browse/NMS-16413) as a placeholder for continuing this work until the refactor is finished.

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16294

[NMS-16413]: https://opennms.atlassian.net/browse/NMS-16413?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by christianpape — APPROVED** [review]
LGTM.

File: pom.xml [hunk 1]
```diff
@@ -1777,7 +1784,7 @@
     <cxfVersion>3.5.5</cxfVersion>
     <cxfXjcVersion>3.3.2</cxfXjcVersion>
     <dhcp4javaVersion>1.1.0</dhcp4javaVersion>
-    <dnsjavaVersion>3.5.3_1</dnsjavaVersion>
+    <dnsjavaVersion>2.1.9_1</dnsjavaVersion>
```
Comment (fooker) [inline pom.xml:1787]: Why do we need to downgrade here? The last release of 2.x was back in 2019 whereas 3.5.3 is the last release available from end of 2023.

**Review by fooker — CHANGES_REQUESTED** [review]
I can not see, why all the DNS code has been changed.

File: pom.xml (hunk 1 above)
Comment (christianpape) [reply pom.xml:1787]: According to mvnrepository.com this seems to be the last version without any CVEs.

File: core/test-api/dns/src/main/java/org/opennms/core/test/dns/DNSServer.java [hunk 2]
```diff
@@ -611,9 +620,9 @@ byte addAnswer(final Message response, final Name name, int type, int dclass, in
     byte[] doAXFR(final Name name, final Message query, final TSIG tsig, TSIGRecord qtsig, final Socket s) {
         final Zone zone = m_znames.get(name);
         boolean first = true;
-        if (zone == null) {
```
[the remark this answers is not in the document]
Comment (RangerRick) [reply core/test-api/dns/src/main/java/org/opennms/core/test/dns/DNSServer.java:614]: same here

File: pom.xml (hunk 1 above)
Comment (RangerRick) [reply pom.xml:1787]: I was having problems with dependencies in Karaf not coming up correctly with the newer DNSJava, I have no idea why. I spent a day trying to figure it out and just said "f**k it, I'll put it back" and then everything started OK. _shrug_

File: features/sentinel/runInPlace.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-test -d repository || (echo "This command must be ran from the features/sentinel directory" && exit 1)
+[ "$(grep -c 'OpenNMS :: Features :: Sentinel' pom.xml)" -eq 1 ] || (echo "This command must be ran from the features/sentinel directory" && exit 1)
```
Comment (synqotik) [inline features/sentinel/runInPlace.sh:4]: `ran` -> `run`

**Review by fooker — APPROVED** [review]

**Pull request merged**